src/shaders/MaterialX/color4.h

```diff
@@ -15,312 +15,267 @@ struct color4
+    return color4 (smoothstep(low.rgb, high.rgb, in.rgb),
+                   smoothstep(low.a, high.a, in.a)
+                   );
 }
 
 color4 smoothstep(float low, float high, color4 in)
 {
-    color4 out = in;
-    out.rgb = smoothstep(low, high, in.rgb);
-    out.a = smoothstep(low, high, in.a);
-    return out;
+    return color4 (smoothstep(low, high, in.rgb),
+                   smoothstep(low, high, in.a)
+                   );
 }
 
 color4 remap(color4 in, color4 inLow, color4 inHigh, color4 outLow, color4 outHigh, int doClamp)
 {
       //remap in from [inLow, inHigh] to [outLow, outHigh], optionally clamping to the new range
 
-      color4 out = in;
-      out.rgb = remap(in.rgb, inLow.rgb, inHigh.rgb, outLow.rgb, outHigh.rgb, doClamp);
-      out.a = remap(in.a, inLow.a, inHigh.a, outLow.a, outHigh.a, doClamp);
-      return out;
+      return color4 (remap(in.rgb, inLow.rgb, inHigh.rgb, outLow.rgb, outHigh.rgb, doClamp),
+                     remap(in.a, inLow.a, inHigh.a, outLow.a, outHigh.a, doClamp)
+                     );
 }
 
 color4 remap(color4 in, float inLow, float inHigh, float outLow, float outHigh, int doClamp)
 {
       //remap in from [inLow, inHigh] to [outLow, outHigh], optionally clamping to the new range
-      color4 out = in;
-      out.rgb = remap(in.rgb, inLow, inHigh, outLow, outHigh, doClamp);
-      out.a = remap(in.a, inLow, inHigh, outLow, outHigh, doClamp);
-      return out;
+      return color4 (remap(in.rgb, inLow, inHigh, outLow, outHigh, doClamp),
+                     remap(in.a, inLow, inHigh, outLow, outHigh, doClamp)
+                     );
 }
 
 color4 fgamma(color4 in, color4 g)
 {
-    color4 out = in;
-    out.rgb = fgamma(in.rgb, g.rgb);
-    out.a = fgamma(in.a, g.a);
-    return out;
+    return color4 (fgamma(in.rgb, g.rgb),
+                   fgamma(in.a, g.a)
+                   );
 }
 
 color4 fgamma(color4 in, float g)
 {
-    color4 out = in;
-    out.rgb = fgamma(in.rgb, g);
-    out.a = fgamma(in.a, g);
-    return out;
+    return color4 (fgamma(in.rgb, g),
+                   fgamma(in.a, g)
+                   );
 }
 
 color4 invert(color4 in, color4 amount)
 {
-    color4 out = in;
-    out.rgb = amount.rgb - in.rgb;
-    out.a = amount.a - in.a;
-    return out;
+    return color4 (amount.rgb - in.rgb,
+                   amount.a - in.a
+                   );
 }
 
 color4 invert(color4 in, float amount)
 {
-    color4 out = in;
-    out.rgb = color(amount) - in.rgb;
-    out.a = amount - in.a;
-    return out;
+    return color4 (color(amount) - in.rgb,
+                   amount - in.a
+                   );
 }
 
 color4 invert(color4 in)
 {
-    color4 out = invert(in, 1.0);
-    return out;
+    return invert(in, 1.0);
 }
 
 color4 clamp(color4 in, color4 low, color4 high)
 {
-    color4 out = in;
-    out.rgb = clamp(in.rgb, low.rgb, high.rgb);
-    out.a = clamp(in.a, low.a, high.a);
-    return out;
+    return color4 (clamp(in.rgb, low.rgb, high.rgb),
```

Comment:
Let me phrase it an alternative way.

For a pixel that's 100% covered (i.e. viewing an opaque object) and white, is the "inverse" a pixel that's covered and black? Or a pixel that's uncovered (completely transparent)?

For a pixel that's empty (rgba==0), what is the inverse? Is it a pixel that's opaque and white? A pixel that's still uncovered, but glowing bright white? Or does invert not change a pixel that "has nothing in it"?

For a pixel that's 3/4 covered with a white thing... so its rgba = (0.75,0.75,0.75,0.75): is its inverse a pixel that/s 3/4 covered with a black thing (0,0,0,.75)?  Or is it a pixel that's (0.25,0.25,0.25,0.25), which is a pixel that's only 1/4 covered, but still by something completely white? That seems wrong? Or something else?

We can certainly table the discussion about what math means for color4 for another PR and another topic. It shouldn't stop you from committing the constructor PR.